packages/cli/src/project-references.test.ts

```diff
@@ -1,22 +1,83 @@
-import { getFixture, getRelativePath } from '@ts-bridge/test-utils';
+import { createDeferredPromise } from '@metamask/utils';
+import { getFixture, getRelativePath, sleep } from '@ts-bridge/test-utils';
 import assert from 'assert';
-import { dirname } from 'path';
+import chalk from 'chalk';
+import { dirname, join } from 'path';
 import type { Program, ResolvedProjectReference } from 'typescript';
 import { factory, ScriptTarget, sys } from 'typescript';
 import { fileURLToPath } from 'url';
-import { beforeAll, describe, expect, it } from 'vitest';
+import { beforeAll, describe, expect, it, vi } from 'vitest';
+import type { WorkerOptions } from 'worker_threads';
 
-import { buildProjectReferences, getProgram } from './build.js';
 import { getTypeScriptConfig } from './config.js';
 import type { DependencyGraph } from './project-references.js';
 import {
+  parallelise,
   createGraph,
   createProjectReferencesCompilerHost,
   getResolvedProjectReferences,
   topologicalSort,
 } from './project-references.js';
+import { buildProjectReferences, getProgram } from './shared.js';
 import { getDefinedArray } from './utils.js';
 
+beforeAll(() => {
+  chalk.level = 0;
+});
+
+vi.mock('worker_threads', async (importOriginal) => {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+  const original = await importOriginal<typeof import('worker_threads')>();
+
+  /**
+   * The worker code that runs TypeScript code. This is executed in the worker by
+   * using it as a data URL.
+   */
+  const WORKER_CODE = `
+    import { createRequire } from 'module';
+    import { workerData } from 'worker_threads';
+
+    const filename = '${import.meta.url}';
+    const tsconfig = '${join(
+      dirname(fileURLToPath(import.meta.url)),
+      '..',
+      'tsconfig.json',
+    )}';
+
+    const require = createRequire(filename);
+    const { tsImport } = require('tsx/esm/api');
+
+    tsImport(workerData.fileName, {
+      parentURL: import.meta.url,
+      tsconfig,
+    });
+  `;
```

Comment:
It is, but I initially couldn't figure out how to share this code, because of this error:

> Error: [vitest] There was an error when mocking a module. If you are using "vi.mock" factory, make sure there are no top level variables inside, since this call is hoisted to top of the file. Read more: https://vitest.dev/api/vi.html#vi-mock

But I just figured out a workaround for this.